### PR TITLE
tests: kernel: Add description for test cases

### PR DESCRIPTION
**Pull request by spoorthik**
Add description to test cases in tests/mem_heap, tests/mem_pool and tests/mem_slab

Signed-off-by: Spoorthi K <spoorthi.k@intel.com>

### CODE REVIEW

File: tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_concept.c [hunk 1]
```diff
@@ -44,7 +54,16 @@ void test_mheap_malloc_align4(void)
 		k_free(block[i]);
 	}
 }
-
```
Comment (pfalcon) [inline tests/kernel/mem_heap/mheap_api_concept/src/test_mheap_concept.c]: This looks like there won't be an empty line between the previous function and the added comment, why this line was removed?

File: tests/kernel/mem_pool/sys_mem_pool/src/main.c [hunk 2]
```diff
@@ -52,7 +55,9 @@ void test_sys_mem_pool_alloc_free(void)
 	/** TESTPOINT: If ptr is NULL, no operation is performed.*/
 	sys_mem_pool_free(NULL);
 }
-
```
Comment (pfalcon) [inline tests/kernel/mem_pool/sys_mem_pool/src/main.c]: Same issue as above.

**Comment by spoorthik** [comment]
@pfalcon Addressed your review comments.

File: tests/kernel/mem_pool/mem_pool_concept/src/test_mpool_alloc_wait.c [hunk 3]
```diff
@@ -65,6 +65,29 @@ void tmpool_alloc_wait_ok(void *p1, void *p2, void *p3)
 }
 
 /*test cases*/
+/**
+ * @brief Test to check the behavior of alloc and free
+ * on memory pool in multiple threads of different priority
+ * scenario.
+ *
+ * @details The test case allocates 3 blocks of MIN_SIZE
```
Comment (nashif) [inline tests/kernel/mem_pool/mem_pool_concept/src/test_mpool_alloc_wait.c]: split this into a few paragraphs please, it is very difficult to read

**Review by nashif — CHANGES_REQUESTED** [review]
Whenever you are testing an API or a feature that is documented somewhere else, use @see <function signature>, <function signature2>, ...

This will allows us to create a trace back to the feature being tested.

**Review by nashif — APPROVED** [review]

**Pull request merged**